lib/db_connection/connection.ex

```diff
@@ -75,6 +75,7 @@ defmodule DBConnection.Connection do
       timer: nil,
       backoff: Backoff.new(opts),
       connection_listeners: Keyword.get(opts, :connection_listeners, []),
+      connection_listeners_tag: Keyword.fetch(opts, :connection_listeners_tag),
```

Comment:
Let's please keep them as a single field since they are always read together!

Reply:
@josevalim what do you have in mind?

  1. `:connection_listeners_with_tag`, which becomes exclusive with `:connection_listeners`
  2. `connection_listeners: [...] | {[...], tag}`
  3. ?

🙃

Reply:
`connection_listeners: [...] | {[...], tag}` or `connection_listeners: {[...], {:ok, tag} | :error}}`